packages/purpose-readmodel-writer/src/purposeConsumerServiceV2.ts

```diff
@@ -43,7 +47,9 @@ export async function handleMessageV2(
           },
           {
             $set: {
-              data: purpose ? fromPurposeV2(purpose) : undefined,
+              data: purpose
+                ? toReadModelPurpose(fromPurposeV2(purpose))
+                : undefined,
```

Comment:
I personally prefer this approach which is cleaner, instead of `adaptPurposeToReadModel` which also wraps the metadata. (I know in catalog there is already `adaptEServiceToReadModel`)

Reply:
I agree, there's no need to also wrap the metadata

Reply:
I have no preference in this regard. Fine for me